larndsim/consts/light.py

```diff
@@ -127,7 +128,7 @@ def set_light_properties(detprop_file):
         TAU_S = float(detprop.get('tau_s', TAU_S))
         TAU_T = float(detprop.get('tau_t', TAU_T))
         
-        LIGHT_GAIN = np.array(detprop.get('light_gain', np.full(OP_CHANNEL_EFFICIENCY.shape, LIGHT_GAIN)))
+        LIGHT_GAIN = np.array(detprop.get('light_gain', [DEFAULT_LIGHT_GAIN]))
         if LIGHT_GAIN.size == 1:
             LIGHT_GAIN = np.full(OP_CHANNEL_EFFICIENCY.shape, LIGHT_GAIN)
         assert LIGHT_GAIN.shape == OP_CHANNEL_EFFICIENCY.shape
```

Comment:
Does L133 not need to now be

`LIGHT_GAIN = np.full(OP_CHANNEL_EFFICIENCY.shape, DEFAULT_LIGHT_GAIN)`

given the repurposing of the `LIGHT_GAIN` variable and inclusion of the new `DEFAULT_LIGHT_GAIN` variable?

Reply:
Ah I see now the np.full can take array-like objects for the fill value, so given line131, these should be equivalent.

Reply:
numpy broadcasting: luxurious convenience for the mere price of making valid code look broken and making code review a massive headache